### ISSUE TITLE
Does emcc support '-fstack-protector'? Error: 'undefined symbol: __stack_chk_guard'

### ISSUE THREAD

**Issue by luopengting** [emscripten-core/emscripten#17030]
Please include the following in your bug report:

**Version of emscripten/emsdk:**
3.1.0
![image](https://user-images.githubusercontent.com/22173722/169978794-1cc866f4-8697-430f-b521-84a6425ed252.png)

**Failing command line in full:**
The compilation script compiles output WASM file and it can run on the web.
```
cmake ..
make
```
If i add `-fstack-protector` in CMakeLists.txt, There are some errors.

![image](https://user-images.githubusercontent.com/22173722/169979069-c4def4db-bc07-4ce3-bbd7-5cd24a8f5109.png)

**Comment by kripken** [comment]
I don't think this is supported atm. However, you can use [sanitizers](https://emscripten.org/docs/debugging/Sanitizers.html), and there is also `STACK_OVERFLOW_CHECK=1` or `2` ([docs](https://emscripten.org/docs/porting/Debugging.html#compiler-settings)) which do related things.
